aten/src/ATen/native/ReduceOps.cpp

```diff
@@ -358,6 +358,9 @@ Tensor& prod_out(Tensor& result, const Tensor& self, Dimname dim,
 
 Tensor &mean_out_cpu_gpu(Tensor &result, const Tensor &self, IntArrayRef dim,
                  bool keepdim, c10::optional<ScalarType> opt_dtype) {
+  auto dim_set = std::set<IntArrayRef::value_type>(dim.cbegin(), dim.cend());
```

Comment:
This is probably not actually the most efficient way to do the duplicate check, since it involves doing a fairly unnecessary dynamic allocation for the set. Probably quickest when number of dims is small (which it should be usually) is just the quadratic nested loops version.

Reply:
Makes sense.
What I had in mind was something like
```c++
if (dim.size() < 10) { // 10 is just some heurestic value
   // for loop version
} else {
  // set version.
}
```
Let me know if it sounds good and if you approve it what should the heuristic value be?

Thank You!